core/src/spotify_id.rs

```diff
@@ -133,3 +255,185 @@ impl fmt::Display for FileId {
         f.write_str(&self.to_base16())
     }
 }
+
+#[inline]
+fn to_base16(src: &[u8], buf: &mut [u8]) -> String {
+    let mut i = 0;
+    for v in src {
+        buf[i] = BASE16_DIGITS[(v >> 4) as usize];
+        buf[i + 1] = BASE16_DIGITS[(v & 0x0f) as usize];
+        i += 2;
+    }
+
+    unsafe {
+        // Safety: We are only dealing with ASCII characters.
+        String::from_utf8_unchecked(buf.to_vec())
+    }
+}
+
+#[cfg(test)]
```

Comment:
I think the added complexity is fine since you've taken the time to add tests.

But could you add running the tests to the travis config?

And if it's not a huge PITA, it would be good to see a commit with the tests passing for the old implementation to prove correctness.